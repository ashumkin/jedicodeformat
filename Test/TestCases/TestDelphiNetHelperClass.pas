unit TestDelphiNetHelperClass;

{
  AFS May 2005
 This unit compiles but is not semantically meaningfull
 it is test cases for the code formatting utility

 Basic test of helper class in Delphi.NET
}


interface

type
TMyClass = class
  public
    procedure HelloMyClass;
  end;

  TMyClassHelper = class helper for TMyClass
  public
    procedure HelloMyClassHelper;virtual;
  end;
 
  TMyClassHelper2 = class helper(TMyClassHelper) for TMyClass
  public
    procedure HelloMyClassHelper;override;
  end;

implementation

{ TMyClass }

procedure TMyClass.HelloMyClass;
begin

end;

{ TMyClassHelper }

procedure TMyClassHelper.HelloMyClassHelper;
begin

end;

{ TMyClassHelper2 }

procedure TMyClassHelper2.HelloMyClassHelper;
begin
  inherited;

end;

end.
