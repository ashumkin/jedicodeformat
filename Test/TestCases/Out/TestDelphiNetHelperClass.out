unit TestDelphiNetHelperClass;

interface

type
  TMyClass = class
  public
    procedure HelloMyClass;
  end;

  TMyClassHelper = class helper for TMyClass
  public
    procedure HelloMyClassHelper; virtual;
  end;

  TMyClassHelper2 = class helper(TMyClassHelper) for TMyClass
  public
    procedure HelloMyClassHelper; override;
  end;

implementation

{ TMyClass }

procedure TMyClass.HelloMyClass;
begin

end;

{ TMyClassHelper }

procedure TMyClassHelper.HelloMyClassHelper;
begin

end;

{ TMyClassHelper2 }

procedure TMyClassHelper2.HelloMyClassHelper;
begin
  inherited;

end;

end.
